core/tests/protractor_utils/CollectionEditorPage.js

```diff
@@ -49,123 +49,122 @@ var CollectionEditorPage = function() {
   };
 
-  this.setCommitMessage = function(message) {
-    waitFor.visibilityOf(saveModal, 'Save Modal takes too long to appear');
-    waitFor.elementToBeClickable(
+  this.setCommitMessage = async function(message) {
+    await waitFor.visibilityOf(
+      saveModal, 'Save Modal takes too long to appear');
+    await waitFor.elementToBeClickable(
       commitMessageInput, 'Commit Message input takes too long to appear');
-    commitMessageInput.click();
-    commitMessageInput.sendKeys(message);
+    await commitMessageInput.click();
+    await commitMessageInput.sendKeys(message);
   };
 
   // Shift a node right in the node graph.
-  this.shiftNodeRight = function(number) {
-    editorShiftRight.get(number).click();
+  this.shiftNodeRight = async function(number) {
+    await editorShiftRight.get(number).click();
   };
 
   // Delete a node in the node graph.
-  this.deleteNode = function(number) {
-    editorDeleteNode.get(number).click();
+  this.deleteNode = async function(number) {
+    await editorDeleteNode.get(number).click();
   };
 
   // Save draft of the collection.
-  this.saveDraft = function() {
-    waitFor.elementToBeClickable(
+  this.saveDraft = async function() {
+    await waitFor.elementToBeClickable(
       saveDraftButton, 'Collection Save Draft button is not clickable');
-    saveDraftButton.click();
+    await saveDraftButton.click();
   };
 
   // Closes the save modal.
-  this.closeSaveModal = function() {
-    waitFor.elementToBeClickable(
+  this.closeSaveModal = async function() {
+    await waitFor.elementToBeClickable(
       closeSaveModalButton, 'Publish Changes button is not clickable');
-    closeSaveModalButton.click();
-    waitFor.invisibilityOf(
+    await closeSaveModalButton.click();
+    await waitFor.invisibilityOf(
       closeSaveModalButton, 'Save Modal takes too long to close');
   };
 
   // Click on publish collection.
-  this.publishCollection = function() {
-    waitFor.elementToBeClickable(
+  this.publishCollection = async function() {
+    await waitFor.elementToBeClickable(
       editorPublishButton, 'Collection Publish button is not clickable');
-    editorPublishButton.click();
+    await editorPublishButton.click();
   };
 
   // Set collection title.
-  this.setTitle = function(title) {
-    editorTitleInput.sendKeys(title);
+  this.setTitle = async function(title) {
+    await editorTitleInput.sendKeys(title);
   };
 
   // Set collection objective.
-  this.setObjective = function(objective) {
-    collectionEditorObjectiveInput.sendKeys(objective);
+  this.setObjective = async function(objective) {
+    await collectionEditorObjectiveInput.sendKeys(objective);
   };
 
   // Set collection category.
-  this.setCategory = function(category) {
-    editorCategoryDropdown.first().click();
-    browser.driver.switchTo().activeElement().sendKeys(category + '\n');
+  this.setCategory = async function(category) {
+    await editorCategoryDropdown.first().click();
```

Comment:
Need nested `await`s here

Reply:
First does not require to be awaited (see [here](https://www.protractortest.org/#/api?view=ElementArrayFinder.prototype.first)).